sa/storage-authority.go

```diff
@@ -165,28 +168,57 @@ func (ssa *SQLStorageAuthority) GetAuthorization(id string) (authz core.Authoriz
 		}
 		authD := authObj.(*authzModel)
 		authz = authD.Authorization
+	}
 
-		err = tx.Commit()
+	var challObjs []challModel
+	_, err = tx.Select(&challObjs, "SELECT * FROM challenges WHERE authorizationID = :authID", map[string]interface{}{"authID": authz.ID})
```

Comment:
Because we currently refer to challenges using the `?challenge=N` query string, the order of challenges within each Authorization must be stable. You should add an explicit `ORDER BY id ASC`.

We may want to give challenges their own independent path one day, but that's for another change.